BinarySearchTree/utils.go

```diff
@@ -0,0 +1,9 @@
+package BinarySearchTree
+
+func keyhash(s string) int {
+	var hash int
```

Comment:
wouldn't there be a problem of integer overflow?
because int is a 32 bit integer and not 64